app/renderer/js/pages/preference/general-section.ts

```diff
@@ -396,6 +404,29 @@ export default class GeneralSection extends BaseSection {
 		});
 	}
 
+	setLocale(): void {
+		const langDiv: HTMLSelectElement = document.querySelector('.lang-div');
+		// This path is for the JSON file that stores key: value pairs for supported locales
+		const path = __dirname.replace('renderer/js/pages/preference', 'translations/supported-locales.json');
+		const data = fs.readFileSync(path, {encoding: 'utf8'});
+		const langs = JSON.parse(data);
+		const langList = this.generateSelectTemplate(langs, 'lang-menu');
+		langDiv.innerHTML += langList;
+		// langMenu is the select-option dropdown menu formed after executing the previous command
+		const langMenu: HTMLSelectElement = document.querySelector('.lang-menu');
+
+		// The next three lines set the selected language visible on the dropdown button
+		let langIndex = ConfigUtil.getConfigItem('languageIndex');
+		langIndex = (langIndex === null) ? 4 : langIndex;
```

Comment:
Storing this as an index is extremely fragile—it means we can never reorder, insert, or delete languages without disturbing other languages.

Reply:
Agreed. Thanks @andersk :)